query/query/query.py

```diff
@@ -29,6 +31,18 @@ def java_to_web_response(jresp):
     return web.json_response(status=status, text=value)
 
 
+async def send_ws_response(thread_pool, ws, f, *args, **kwargs):
+    jresp = await blocking_to_async(thread_pool, f, *args, **kwargs)
+    status = jresp.status()
+    value = jresp.value()
+    log.info(f'response status {status} value {value}')
+    if status in (400, 500):
+        await ws.send_json({'error': value})
+    else:
+        assert status == 200, status
+        await ws.send_json({'status': status, 'result': value})
```

Comment:
I would prefer a consistent wire format. Let's go with `{'status': int, 'result': optional[any]}` and result may be elided instead of setting it to `null`.

Reply:
In particular, I don't like having `'error'` be a distinct field name.